server/controllers/finance/feeCenter.js

```diff
@@ -0,0 +1,230 @@
+      res.status(201).json({ id: result.insertId });
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Update a fee center in the database
+*
+* @example
+* // PUT /fee_centers : update a fee center
+* var feeCenters = require('finance/fee_center');
+* feeCenters.update(req, res, next);
+*/
+
+function update (req, res, next) {
+  'use strict';
+
+  var queryData = req.body;
+  var feeCenterId = req.params.id;
+  var updateFeeCenterQuery = 'UPDATE fee_center SET ? WHERE id = ?';
+
+  delete queryData.id;
+
+
+   db.exec(updateFeeCenterQuery, [queryData, feeCenterId])
+    .then(function (res) {
+
+      if(res.affectedRows === 0){
+        throw new NotFound(`Could not update a fee center with id ${feeCenterId}`);
+      }
+
+      return lookupFeeCenter(feeCenterId);
+    })
+    .then(function (feeCenter) {
+      res.status(200).json(feeCenter);
+    })
+    .catch(next)
+    .done();
+}
+
+
+/**
+* Remove a fee center in the database
+*
+* @example
+* // DELETE /fee_centers : delete a fee center
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.remove(req, res, next);
+*/
+
+function remove (req, res, next) {
+  var feeCenterId = req.params.id;
+  var removeFeeCenterQuery = 'DELETE FROM fee_center WHERE id = ?';
+
+  return db.exec(removeFeeCenterQuery, [feeCenterId])
+    .then(function (res) {
+      if(res.affectedRows === 0){
+        throw new NotFound(`Could not remove a fee center with id ${feeCenterId}`);
+      }
+      res.sendStatus(204);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Return a fee center details from the database
+* @example
+* // GET /fee_centers : returns a fee center detail
+* var feeCenters = require('finance/fee_centers');
+* feeCenters.detail(req, res, next);
+*/
+
+function detail(req, res, next) {
+  'use strict';
+
+  lookupFeeCenter(req.params.id)
+    .then(function (row) {
+      res.status(200).json(row);
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+* Return a fee center instance from the database
+*
+* @param {integer} id of a fee center
+*
+*/
+
+function lookupFeeCenter(id) {
+  'use strict';
+
+  var sql =
+    'SELECT fc.id, fc.label, fc.is_cost, fc.note, fc.is_principal, fc.project_id FROM fee_center AS fc WHERE fc.id = ?';
+
+  return db.one(sql, id)
+    .then(function (rows) {
+      if (rows.length === 0) {
+        throw new NotFound(`Could not find a fee center with id ${id}`);
+      }
+      return rows;
+    });
+}
+
+/**
+* Return a fee cost value by scanning the general ledger
+*
+*
+* @example
+* // GET /fee_centers/:id/value : returns a value of the fee center
+* var feeCenters = require('finance/fee_center');
+* feeCenters.getFeeValue(req, res, next);
+*/
+
+function getFeeValue (req, res, next){
+  var sql = null;
+
+
+  lookupFeeCenter(req.params.id)
+    .then(function (){
+
+      sql =
+        `
+        SELECT
+          IFNULL(IF(t.is_cost = 1, SUM(t.debit_equiv - t.credit_equiv), SUM(t.credit_equiv - t.debit_equiv)), 0) AS value, t.id
```

Comment:
This is a pretty complicated query to understand.  It would be more readable if you did the `SUM()` and `IF()` functions on different statements.

Alternatively, you could just do a multiplication by `-1` to inverse the amounts, with a comment explaining why it should be inverted.
```js
const sql = `
  SELECT id, IF(is_cost = 1, value, value * -1) AS value
  FROM (
    SELECT SUM(gl.debit_equiv - gl.credit_equiv) AS value, f.is_cost, f.id
    FROM general_ledger AS gl JOIN fee_center AS f
      ON gl.fc_id = f.id
    WHERE gl.fc_id = ?
    GROUP BY gl.fc_id
  )z;
`
```
This query may also be more performant by only using a single `SUM()`.